homeassistant/components/modbus/entity.py

```diff
@@ -177,9 +177,18 @@ def async_hold(self) -> None:
         self._attr_available = False
         self.async_write_ha_state()
 
+    async def async_await_connection(self, _now: Any) -> None:
+        """Wait for first connect."""
+        await self._hub.event_connected.wait()
+        self.async_run()
+
     async def async_base_added_to_hass(self) -> None:
         """Handle entity which will be added."""
-        self.async_run()
+        async_call_later(
+            self.hass,
+            self._hub.config_delay + 0.1,
+            self.async_await_connection,
+        )
```

Comment:
When does this execute? In theory this also is a listener that we should cancel when the entity unloads

Reply:
self._cancel_timer and self._cancel_call are examples of other hanging "things".

To be quite honest, I have been pulling hair out (that I don't have) because there are so many forgotten details.

Anyhow you have a point and I will a variable to make sure it is not forgotten later.

Reply:
I am not sure why we need this in a variable, you can just do `self.async_on_remove` right?

Reply:
Did not know that one, let me try it out.

Reply:
Ups, really got me there !!!!!